### DISCUSSION TITLE
Issue with z_offset

### DISCUSSION THREAD

**Discussion by dpeart** [jschuh/klipper-macros#181] [Q&A]
I finally tracked my issue to improper z_offset down to my loading these macros.  If I don't load them I perfect first layer printing, if I do load them my first layer printing goes to crap.

It acts as if the z_offset is totally messed up. I can even adjust the z_offest as it is printing and it doesn't affect the printer at all.  I change my z_offset in the config and nothing.

The only way I can get the printer to work is unload these macros.

I want to use them as they provide a lot of functionality I like, but I need to figure out why it breaks my printer.

Voron 2.4 R2

Suggestions?

**Comment by dpeart** [comment]
Ok I re-read the installation instructions again, and as was suggested in a nother conversation here, I removed the mainsail.cfg.  It works now, so something conflicted between the two.

thanks

**Comment by dpeart** [comment]
OK I'm missing something.
It worked for a coupte of prints, then it stopped.

It appears to ignore my printer.cfg  z_offset setting completely.  If I interactively change it after the print has started, and set it to what the printer.cfg says then it prints nicely.  Not ignoring, it must overwrite it because when I don't load these macros it works fine.

I don't load mainsail.cfg, just my printer.cfg, this collection of macros and my personal nozzle cleaner macro.

**Comment by dpeart** [comment]
I think this has to do with bed_surfaces.  The setting in variable.cfg seems to override my printer.cfg.

How should this be used, or how do I disable it?

**Comment by PaleSasquatch** [comment]
I think, i have the same problem. The Z offset changes when i start the print.

**Reply by dpeart** [reply]
I just renamed the klipper-macros/bed_surface.cfg and added a .bak to it.  That way that functionality isn't ever loaded and it has worked fine for me.

I don't think there is a way to configure it to turn it off.  I do get an error _LOAD_INIT_SURFACE or something but it doesn't hurt things.

Honestly, I'm in the process of moving to Kipplain instead.

**Reply by ovar0409** [reply]
It seems I have the same problem. I'm not able to understand what those macros are doing regarding z offset calibration
I tried to redo a PROBE_CALIBRATE with the paper test but then it produces a negative value z-offset. Of course if I save and restart, Klipper complains and asks to fix this error...

I don't understand how to first calibrate the z-offset when the macros are loaded...

I can use  SET_SURFACE_OFFSET to set the offset to the same value as the one I had (using PROBE_CALIBRATE  without those macros loaded), and I noticed that when using the Mainsail Z-offset controls, using ADJUST_SURFACE_OFFSETS would indeed adjust the z-offset for this bed surface (check done via  SET_SURFACE_OFFSET without passing a value), but how does it get the initial value of the z-offset ?
